phpdotnet/phd/IndexRepository.php

```diff
@@ -189,13 +189,7 @@ public function getRefNames(): array {
 
 
     private function SQLiteRefname($context, $index, $id, $sdesc): void {
-        $ref = strtolower(
-                str_replace(
-                    ["_", "::", "->"],
-                    ["-", "-", "-"],
-                    html_entity_decode($sdesc, ENT_QUOTES, 'UTF-8')
-                )
-            );
+        $ref = strtolower(html_entity_decode($sdesc, ENT_QUOTES, 'UTF-8'));
```

Comment:
Should there be a "normalisation" function that we can use in the different packages?

Reply:
I'm not sure. As you can see from my changes there are three places where these references are accessed (PHP XHTML and PDF rendering, and this where the indexes are retrieved from the database) but this instance doesn't match the other two. All three operate on the same XML document and access the same `xml:id`s on the same `<refentry>`'s. There's no issues with linking these functions/methods because (I think) this `html_entity_decode()` doesn't do anything on our valid `xml:id`s anyway.

TL;DR this line in `IndexRepository.php` should probably be `$ref = strtolower($sdesc);` like the other two lines and there's probably no normalization function needed.